backend/src/routes/v0/schedule.py

```
@@ -136,3 +262,43 @@ async def delete_job(
+    response_model=list[ScheduleExecutionResponse],
+    operation_id="ruska_list_schedule_executions",
+)
+async def get_schedule_executions(
+    schedule_id: str,
+    user: ProtectedUser = Depends(verify_credentials),
+):
+    # Verify schedule exists and belongs to user (raises 403 if not)
+    try:
+        schedule_service.user_id = user.id
+        schedule_service.get_job(schedule_id)
+    except Exception:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)